src/components/Aframe/my-loader.js

```diff
@@ -68,27 +68,26 @@ AFRAME.registerComponent("myloader", {
     this.onSelectStart = this.onSelectStart.bind(this);
     this.onClearCollide = this.onClearCollide.bind(this);
     this.loadModel = this.loadModel.bind(this);
-    this.updateTransfertexture = this.updateTransfertexture.bind(this);
+    this.updateTransferTexture = this.updateTransferTexture.bind(this);
     this.updateColorMapping = this.updateColorMapping.bind(this);
     this.debugScene = this.debugScene.bind(this);
 
+    this.updateOpacityData = this.updateOpacityData.bind(this);
+    this.colorMapNeedsUpdate = false;
+    this.currentColorMap = "";
+
     //window.addEventListener('keydown', this.debugScene);
     this.el.addEventListener("raycaster-intersected", this.onCollide);
     this.el.addEventListener(
       "raycaster-intersected-cleared",
       this.onClearCollide
     );
-    this.colorTransfer = new Uint8Array(3 * 256);
+    //this.colorTransfer = new Uint8Array(3 * 256);
+    this.colorTransferMap = new Map();
 
     this.group = new THREE.Group();
-    //this.el.sceneEl.object3D.add(group);
-    this.colorMap = {
-      img: null,
-      data: null,
-    };
 
-    this.colorMap.img = document.createElement("img");
-    this.colorTransfer = new Uint8Array(3 * 256);
+    //this.colorTransfer = new Uint8Array(3 * 256);
```

Comment:
delete commented out code (here and elsewhere)

Reply:
done